src/interface/src/app/plan/plan.component.ts

```diff
@@ -1,13 +1,75 @@
-import { Component } from '@angular/core';
+import { AfterViewInit, Component, OnDestroy } from '@angular/core';
 import { Router } from '@angular/router';
+import * as L from 'leaflet';
+
+import { Plan, Region } from '../types';
 
 @Component({
   selector: 'app-plan',
   templateUrl: './plan.component.html',
   styleUrls: ['./plan.component.scss'],
 })
-export class PlanComponent {
-  constructor(private router: Router) {}
+export class PlanComponent implements AfterViewInit, OnDestroy {
+  map!: L.Map;
```

Comment:
Do we want to try making the PlanComponent the parent component and have the Map and Panels as child components instead of making it one giant component like the explore Map? Organization and modularity-wise it seems to make sense to split them out. The simple components just have input and output, while the more complex ones can use the state services to pass data. Maybe we need a shared state service just for this Plan component, since it's so big and has a lot going on.. like the summary, adding priorities, scenarios, comments, etc.

Reply:
That's a good idea! I'll work on that.

Reply:
Okay, I made the map its own child component!

Reply:
Yay! Looks great :)